src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -2174,6 +2180,11 @@ internal PackageInfo GetNodePackage(NodeModel node)
                         throw new Exception("There are multiple subscribers to Workspace.CollectingNodePackageDependencies. " +
                             "Only PackageManagerExtension should subscribe to this event.");
                     }
+                    //in case of python node, return python engine dependency
+                    if (node.IsPython && CollectingPythonEnginePackageDependencies != null)
```

Comment:
Why can't we just check for inheritance of PythonNodeBase?

Reply:
I guess to allow other base classes? Just not sure how likely that is, and it feels super specific to include in `NodeModel`.

Reply:
That is what I wanted to do as well, but it will introduce a PythonNodeModel project dependency in Core, so I refrained from that. But can go that route as well, thoughts? @pinzart90